Plugins/Public/base_plugin/PlayerBase.cpp

```diff
@@ -664,15 +651,71 @@ void ReportAttack(wstring basename, wstring charname, uint system)
 	return;
 }
 
+// For all players in the base's system, resync their reps towards all objects
+// of this base.
+void PlayerBase::SiegeModChainReaction(uint client)
+{
+	map<uint, PlayerBase*>::iterator it;
+	for (it = player_bases.begin(); it != player_bases.end(); it++)
+	{
+		if (it->second->system == this->system)
+		{
+			if (HkDistance3D(it->second->position, this->position) < siege_mod_chain_reaction_trigger_distance)
+			{
+				if (!(it->second->siege_mode))
+				{
+					float attitude = it->second->GetAttitudeTowardsClient(client, true);
+					if (attitude < -0.55f)
+					{
+						it->second->siege_mode = true;
+
+						const wstring& charname = (const wchar_t*)Players.GetActiveCharacterName(client);
+						ReportAttack(it->second->basename, charname, it->second->system, L"has detected hostile activity at the nearby base by");
+
+						it->second->SyncReputationForBase();
+					}
+				}
+			}
+		}
+	}
+}
+
+// For all players in the base's system, resync their reps towards this object.
+void PlayerBase::SyncReputationForBaseObject(uint space_obj)
```

Comment:
please can we put this function and its comment back where we found it?

Reply:
If you insist